sdk/healthinsights/health-insights-clinicalmatching-rest/samples-dev/sample_match_trials_fhir.ts

```diff
@@ -0,0 +1,155 @@
+            console.log(`Type: ${String(tmInferences.type)}  Value: ${tmInferences.value}`);
+            console.log(`Description ${tmInferences.description}`);
+          }
+        }
+      }
+    }
+    else {
+      const errors = trialMatcherResult.errors;
+      if (errors) {
+          for (const error of errors) {
+              console.log(error.code, ":", error.message);
+          }
+      }
+  }
+}
+
+export async function main() {
+  const credential = new AzureKeyCredential(apiKey);
+  const client = ClinicalMatchingRestClient(endpoint, credential);
+
+  const clinicalInfoList = [
+    {
+      system: "http://www.nlm.nih.gov/research/umls",
+      code: "C0006826",
+      name: "Malignant Neoplasms",
+      value: "true",
+    },
+    {
+      system: "http://www.nlm.nih.gov/research/umls",
+      code: "C1522449",
+      name: "Therapeutic radiology procedure",
+      value: "true",
+    },
+    {
+      system: "http://www.nlm.nih.gov/research/umls",
+      code: "C1512162",
+      name: "Eastern Cooperative Oncology Group",
+      value: "1",
+    },
+    {
+      system: "http://www.nlm.nih.gov/research/umls",
+      code: "C0019693",
+      name: "HIV Infections",
+      value: "false",
+    },
+    {
+      system: "http://www.nlm.nih.gov/research/umls",
+      code: "C1300072",
+      name: "Tumor stage",
+      value: "2",
+    }
+  ];
+
+  const patientInfo = {
+    sex: "MALE",
+    birthDate: new Date(1965, 11, 26), // Note: Months are zero-based (11 represents December)
+    clinicalInfo: clinicalInfoList,
+  };
+
+  const docContent = {sourceType: "INLINE", value: getPatientDocContent()};
+  const patientDataList = {
+      type: "fhirBundle",
+      id: "Consultation-14-Demo",
+      content: docContent,
+      clinicalType: "CONSULTATION"
+  };
+
+  const patient1 = {
+    id: "patient_id",
+    info: patientInfo,
+    data: [patientDataList]
+  };
+
+  const geographicLocation = { countryOrRegion: "United States", city: "Gilbert", state: "Arizona" };
+  const registryFilters = {
+    conditions: ["Non-small cell lung cancer"],
+    phases: ["PHASE1"],
+    sources: ["CLINICALTRIALS_GOV"],
+    facilityLocations: [ geographicLocation ],
+    studyTypes: ["INTERVENTIONAL"]
+  };
+
+  const clinicalTrials = ({
+    registryFilters: [registryFilters]
+  });
+
+  const configuration = {
+    clinicalTrials: clinicalTrials,
+  };
+
+  const trialMatcherData = {
+    patients: [patient1],
+    configuration: configuration,
+  };
+
+  const trialMatcherParameter = {
+    body: trialMatcherData
+  };
+
+  const initialResponse = await client.path("/trialmatcher/jobs").post(trialMatcherParameter);
+  if (isUnexpected(initialResponse)) {
+    throw initialResponse;
+  }
+  const poller = await getLongRunningPoller(client, initialResponse);
+  const res = await poller.pollUntilDone();
+  if (isUnexpected(res)) {
+      throw initialResponse;
+  }
+  const resultBody = res.body as TrialMatcherResultOutput;
```

Comment:
what's the type we get for res.body ? Just wonder if the type conversion is necessary.

Reply:
res is  type HttpResponse and res.body is unknown. So, it is necessary to use type conversion.